### ISSUE TITLE
Snapcraft: Lost flags or naming ambiguitiy

### ISSUE THREAD

**Issue by jaredfolkins** [goreleaser/goreleaser#1263]
First off, thanks for your work! I appreciate it very much! Below is the weird behavior I'm finding.

**Describe the bug**

When using a single binary with multiple snaps, there seems to be a lot of naming ambiguity or lost flags.

**To Reproduce**

Use kushtakad to test

```
$ git clone https://githug.com/kushtaka/kushtakad
$ cd kushtakad
$ goreleaser --snapshot --skip-publish --rm-dist
$ cat dist/kushtakad_sensor_v0.2.2-beta_Tux_64-bit/prime/meta/snap.yaml 
name: kushtakad-sensor
version: v0.2.2-beta
summary: Kushtakad Sensor
description: Kushtaka.org is a sustainable all-in-one honeypot and honeytoken orchestrator
  for under-resourced Blue Teams.
license: Apache-2.0
grade: devel
confinement: devmode
architectures:
- amd64
apps:
  kushtakad:
    command: kushtakad -sensor
    plugs:
    - network
    - network-bind
    daemon: simple
  kushtakad-sensor:
    command: kushtakad -sensor
    plugs:
    - network
    - network-bind
    daemon: simple

```

**Expected behavior**

I would expect it to only have one appL in the snap.yaml file and yet there are two. This also clutters up the UX for the user when running `snap` for that config, why two?

```
$ sudo snap install --devmode ./dist/kushtakad_sensor_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-sensor v0.2.2-beta installed
$ sudo snap install --devmode ./dist/kushtakad_server_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-server v0.2.2-beta installed
$ sudo snap services
Service                            Startup  Current   Notes
kushtakad-sensor.kushtakad         enabled  inactive  -
kushtakad-sensor.kushtakad-sensor  enabled  inactive  -
kushtakad-server.kushtakad         enabled  inactive  -
kushtakad-server.kushtakad-server  enabled  active    -
```
Also, It would be ideal if I could make it so that the name of the snap was `kushtakad-sensor` and the name of the app: was `daemon` as that is how the examples on snapcraft.io show it.

```
$ sudo snap install --devmode ./dist/kushtakad_sensor_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-sensor v0.2.2-beta installed
$ sudo snap install --devmode ./dist/kushtakad_server_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-server v0.2.2-beta installed
$ sudo snap services
Service                            Startup  Current   Notes
kushtakad-sensor.daemon        enabled  inactive  -
kushtakad-server.daemon         enabled  active    -
```
**Environment (please complete the following information):**

- OS: ubuntu 18.04
- OS version: 5.0.0-36-generic #39~18.04.1-Ubuntu SMP Tue Nov 12 11:09:50 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
- GoReleaser Version 
version: 0.123.3
commit: 36126ec4869a93ef82895b225ee3d761fd87ae15
built at: 2019-11-24T14:40:57Z
built by: goreleaser

**Additional context**
Here is the area of my goreleaser.yml file where I configure the snapcraft apps directive.

https://github.com/kushtaka/kushtakad/blob/master/.goreleaser.yml#L54-L107

**Comment by jaredfolkins** [comment]
Alright I was able to get it to work with the following. 

- I had to compile 2 extra binaries with the names I wanted to set in the snapcraft.yaml files. 
- I had to then exclude those binaries from the archive with the builds directive.
- I also had to add build directives to the apps: sections for each snap

```
$ sudo snap install --devmode dist/kushtakad_sensor_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-sensor v0.2.2-beta installed
$ sudo snap install --devmode dist/kushtakad_server_v0.2.2-beta_Tux_64-bit.snap 
kushtakad-server v0.2.2-beta installed
$ sudo snap services
Service                            Startup  Current   Notes
kushtakad-sensor.kushtakad-sensor  enabled  inactive  -
kushtakad-server.kushtakad-server  enabled  active    -
$ cat .goreleaser.yml 
##
## BUILDS
## 
builds:
##
## kushtakad
##
- id: kushtakad
  main: ./main.go
  binary: kushtakad
  goos:
    - darwin
    - linux
    - windows
  goarch:
    - amd64
  env:
    - GO111MODULE=on
    - CGO_ENABLED=0

- id: kushtakad-server
  main: ./main.go
  binary: kushtakad-server
  goos:
    - darwin
    - linux
    - windows
  goarch:
    - amd64
  env:
    - GO111MODULE=on
    - CGO_ENABLED=0

- id: kushtakad-sensor
  main: ./main.go
  binary: kushtakad-sensor
  goos:
    - darwin
    - linux
    - windows
  goarch:
    - amd64
  env:
    - GO111MODULE=on
    - CGO_ENABLED=0

##
## BEFORE
## 
before:
  hooks:
    - go mod download
    - packr2 clean
    - packr2
      
checksum:
  name_template: 'checksums.txt'

snapshot:
  name_template: "{{ .Tag }}-beta"

changelog:
  sort: asc

##
## ARCHIVES
##
archives:
  ##
  ## kushtakad
  ##
  - id: kushtakad
    builds:
    - kushtakad
    format: gz
    format_overrides:
      - goos: windows
        format: zip
    files:
    - none*

##
## SNAPS
##
snapcrafts:
  - id: kushtakad-server
    builds:
    - kushtakad-server

    name_template: "{{ .ProjectName }}_server_{{ .Version }}_{{ .Os }}_{{ .Arch }}"

    replacements:
      amd64: 64-bit
      darwin: macOS
      linux: Tux

    name: kushtakad-server
    summary: Kushtakad Server
    description: Kushtaka.org is a sustainable all-in-one honeypot and honeytoken orchestrator for under-resourced Blue Teams. 

    license: Apache-2.0
    publish: false
    grade: devel
    confinement: devmode
    #grade: edge
    #base: core18
    #confinement: classic
    apps:
      kushtakad-server:
        plugs: ["network", "network-bind"]
        daemon: simple

  - id: kushtakad-sensor
    builds:
    - kushtakad-sensor

    name_template: "{{ .ProjectName }}_sensor_{{ .Version }}_{{ .Os }}_{{ .Arch }}"

    replacements:
      amd64: 64-bit
      darwin: macOS
      linux: Tux

    name: kushtakad-sensor
    summary: Kushtakad Sensor
    description: Kushtaka.org is a sustainable all-in-one honeypot and honeytoken orchestrator for under-resourced Blue Teams. 

    license: Apache-2.0
    publish: false
    grade: devel
    confinement: devmode
    #base: core18
    #grade: edge
    #confinement: classic
    apps:
      kushtakad-sensor:
        plugs: ["network", "network-bind"]
        daemon: simple
        args: -sensor



```

**Comment by jaredfolkins** [comment]
You can also see here there are no longer multiple app directives. I do believe this is a bug and the ambiguity will hurt goreleaser's use with snaps for some use cases. Also, in some cases it will drop the `-sensor` flag if I don't name: the snap exactly the same as the binary. FWIW. 

```
$  cat dist/kushtakad_server_v0.2.2-beta_Tux_64-bit/prime/meta/snap.yaml 
name: kushtakad-server
version: v0.2.2-beta
summary: Kushtakad Server
description: Kushtaka.org is a sustainable all-in-one honeypot and honeytoken orchestrator
  for under-resourced Blue Teams.
license: Apache-2.0
grade: devel
confinement: devmode
architectures:
- amd64
apps:
  kushtakad-server:
    command: kushtakad-server
    plugs:
    - network
    - network-bind
    daemon: simple            
$ cat dist/kushtakad_sensor_v0.2.2-beta_Tux_64-bit/prime/meta/snap.yaml 
name: kushtakad-sensor
version: v0.2.2-beta
summary: Kushtakad Sensor
description: Kushtaka.org is a sustainable all-in-one honeypot and honeytoken orchestrator
  for under-resourced Blue Teams.
license: Apache-2.0
grade: devel
confinement: devmode
architectures:
- amd64
apps:
  kushtakad-sensor:
    command: kushtakad-sensor -sensor
    plugs:
    - network
    - network-bind
    daemon: simple
```

**Comment by caarlos0** [comment]
fixed by  #1267?

**Comment by caarlos0** [comment]
closing, please reopen/comment if needed @jaredfolkins
